packages/metrics/src/DimensionsStore.ts

```diff
@@ -0,0 +1,70 @@
+import { InvokeStore } from '@aws/lambda-invoke-store';
+import type { Dimensions } from './types/Metrics.js';
+
+/**
+ * Manages storage of metrics dimensions with automatic context detection.
+ *
+ * This class abstracts the storage mechanism for metrics, automatically
+ * choosing between AsyncLocalStorage (when in async context) and a fallback
+ * object (when outside async context). The decision is made at runtime on
+ * every method call to support Lambda's transition to async contexts.
+ */
+class DimensionsStore {
```

Comment:
I suspect we could create a single store abstraction and have each of these stores implement a facade pattern but for the moment I want to keep it simple. Premature abstractions can cause trouble.